docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/FastTree.cs

```diff
@@ -45,11 +75,21 @@ private static IEnumerable<DataPoint> GenerateRandomDataPoints(int count)
             }
         }
 
+        // Example with label and 50 feature values. A data set is a collection of such examples.
         private class DataPoint
         {
             public float Label { get; set; }
             [VectorType(50)]
             public float[] Features { get; set; }
         }
+
+        // Class used to capture predictions.
+        private class Prediction
+        {
+            // Original label.
+            public float Label { get; set; }
+            // Predicted score from the trainer.
+            public float Score { get; set; }
+        }
```

Comment:
I like extending the DataPoint class, because that is effectively what happens, columns get added. not a bigge though.

Reply:
There's cons and pros here. I think keeping input and output separate is easier to understand for users. 

---
In reply to: [266150723](https://github.com/dotnet/machinelearning/pull/2979#discussion_r266150723) [](ancestors = 266150723)